src/utils/generateEventLink.ts

```diff
@@ -0,0 +1,102 @@
+import {
+    CalendarEvent as CalendarLinkEvent,
+    google,
+    outlook,
+    ics,
+} from 'calendar-link';
+import { parse } from 'date-fns';
+
+export const enum CalendarType {
+    Google,
+    Outlook,
+    ICS,
+}
+
+export interface CalendarEventDate {
+    day: number;
+    month: number;
+    year: number;
+    optional?: {
+        hours: number;
+        minutes: number;
+    };
+}
+
+export interface CalendarEvent {
+    title: string;
+    allDay: boolean;
+    location: string;
+    start: CalendarEventDate;
+    end?: CalendarEventDate;
+}
+
+/**
+ * Generates calendar link for an event by a calendar type
+ * @param event Event
+ * @param calendarType CalendarType
+ */
+export function generateEventLink(
+    event: CalendarEvent,
+    calendarType: CalendarType
+): string {
+    const calendarEvent = generateCalendarEvent(event);
+    switch (calendarType) {
+        case CalendarType.Google:
+            return google(calendarEvent);
+        case CalendarType.Outlook:
+            return outlook(calendarEvent);
+        default:
+            return decodeURIComponent(
+                ics(calendarEvent).split('charset=utf8,')[1]
+            );
+    }
+}
+
+/**
+ * in case of changing
+ * update `parseEventDatesIntoSpecialFormat` function
+ */
+const FORMAT = 'd/M/yyyy/H/m';
+
+/**
+ * Transforms Event interface into CalendarEvent from `calendar-link`
+ * @param event CalendarEvent
+ */
+function generateCalendarEvent(event: CalendarEvent): CalendarLinkEvent {
+    const start = parse(
```

Comment:
May you please elaborate on what is achieved here? I think it will be giving us incorrect results, because if we use `new Date()` as a reference date, then the reference date will be always one which is relevant for the server on which the script runs, but not relevant for the user's time zone. I think it's caused the issues with the pipeline.

Reply:
good points, I will double-check it